bio2zarr/writer.py

```diff
@@ -615,13 +160,14 @@ class VcfZarrWriteSummary(core.JsonDataclass):
 
 
 class VcfZarrWriter:
-    def __init__(self, path):
+    def __init__(self, source_type, path):
```

Comment:
Why does the writer need to know the class of the source ahead of init?

Reply:
`load_metadata` needs it, which can be called by e.g. `encode_parition` without `init` being called.